mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2842,8 +2851,18 @@ static LogicalResult produceSubViewErrorMsg(SliceVerificationResult result,
   llvm_unreachable("unexpected subview verification result");
 }
 
-/// Verifier for SubViewOp.
 LogicalResult SubViewOp::verify() {
+  for (int64_t offset : getStaticOffsets()) {
```

Comment:
Can we move those checks to `mlir::detail::verifyOffsetSizeAndStrideOp` and mention in the interface description that offsets and sizes must be non-negative?

Reply:
Yes that makes a lot of sense IMO. I find it quite impressive from the MLIR developers that adding the check (e4e40f2) causes none of the tests to fail! Hopefully nobody downstream depends on this logic even though it shouldn't work anyway.